script.js

```diff
@@ -1,12 +1,210 @@
+let _localEpisodes = [];
+let _localShows = [];
+let _showDDLCtrlId = "selShow";
+let _epDDLCtrlId = "selName";
+function sortByName(arr) {
+  return arr.sort(function (a, b) {
+    var textA = a.name.toLowerCase();
+    var textB = b.name.toLowerCase();
+    return textA < textB ? -1 : textA > textB ? 1 : 0;
+  });
+}
+async function getAllShowAsync() {
+  if (_localShows.length == 0) {
+    _localShows = sortByName(getAllShows());
+  }
+  return true;
+}
+async function getAllEpisodesAsync(showId) {
+  if (!showId && _localShows.length == 0) {
+    console.error("Cannot get shows");
+  }
+  let currentShowId = showId ?? _localShows[0].id;
+  let localShowsEpisodes = {};
+  //use local storage to avoid api load
+  let localShowsEpisodesJsonStr = localStorage.getItem("localShowsEpisodes");
```

Comment:
nice job experimenting with using local storage to store data - have a think about if this has any drawbacks - for example, what if the data from the api was updated/changed? 
You can also just save your data to a variable (and use this variable whenever a user searches for an episode, rather than making a new api request.) It's good that you're starting to think about ways to reduce unnecessary api requests.